src/diagram/diagram-api.tsx

```diff
@@ -156,15 +280,33 @@ export interface DiagramCallbacks {
 
   onDeleteEdge: (identifier: string) => void;
 
+
   // Selection
 
+  /**
+   * This property stores the method, which is called when user changes the selection.
+   * Callback is registered on both node and edge selection.
+   * @param nodes are identifiers of the nodes representing the new node selection.
+   * @param edges are identifiers of the edges representing the new edge selection.
+   */
   onSelectionDidChange: (nodes: string[], edges: string[]) => void;
 
+
   // Connections
 
+  /**
+   * This property stores the method, which is called when user creates connection inside diagram.
+   * @param sourceIdentifier is the identifier of the source node of the connection.
+   * @param targetIdentifier is the identifier of the target node of the connection.
+   */
   onCreateConnectionToNode: (sourceIdentifier: string, targetIdentifier: string) => void;
 
-  onCreateConnectionToNothing: (sourceIdentifier: string, position: Position) => void;
-
 
+  /**
+   * This property stores the method, which is called when user creates "empty" connection,
+   * i. e. connection from node to canvas.
+   * @param sourceIdentifier is the identifier of the node, where the connection started.
+   * @param position is the position on canvas, where the connection ended.
+   */
+  onCreateConnectionToNothing: (sourceIdentifier: string, position: Position) => void;
```

Comment:
In Reactflow it depends on the starting handle of the connection. So if we start the connection from the target handle the connection is actually reversed. So maybe consider if this method shouldn't have some boolean property saying if the `sourceIdentifier` is actually going from the source handle or the target handle

Reply:
In general I do not like booleans, instead I would use enum. On the other hand I think the fact that we have source and target should not matter. We should allow user to draw a connection from one component (source) to another (target).